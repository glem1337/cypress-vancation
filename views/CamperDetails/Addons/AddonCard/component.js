import React from 'react';
import { Button, Tag } from 'antd';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { MAX_DESCRIPTION_VISIBLE_SYMBOLS } from 'constants/camperDetails/addons';

const AddonCard = ({
  iconUrl,
  price,
  name,
  priceUnit,
  description,
  visibleDescription,
  allDescriptionVisible,
  toggleDescription,
}) => (
  <div
    className={classNames('van-details__adds-card', {
      'h-auto': allDescriptionVisible,
    })}
  >
    <div className="d-flex align-items-center justify-content-between mb-20">
      {iconUrl && <img src={iconUrl} alt="" />}
      <Tag color="default">{`$${price} ${priceUnit}`}</Tag>
    </div>
    <p className="text-subtitle font-700 mb-8">{name}</p>
    <p>{visibleDescription}</p>
    {description.length > MAX_DESCRIPTION_VISIBLE_SYMBOLS && (
      <Button onClick={toggleDescription} type="simple-text">
        <FormattedMessage
          id={allDescriptionVisible ? 'shared.hide' : 'shared.readMore'}
        />
      </Button>
    )}
  </div>
);

AddonCard.defaultProps = {
  iconUrl: undefined,
};

AddonCard.propTypes = {
  iconUrl: PropTypes.string,
  price: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  priceUnit: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  visibleDescription: PropTypes.string.isRequired,
  allDescriptionVisible: PropTypes.bool.isRequired,
  toggleDescription: PropTypes.func.isRequired,
};

export default AddonCard;
