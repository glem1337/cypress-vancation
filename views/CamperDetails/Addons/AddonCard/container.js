import React from 'react';
import PropTypes from 'prop-types';

import { MAX_DESCRIPTION_VISIBLE_SYMBOLS } from 'constants/camperDetails/addons';

import AddonCardComponent from './component';

class AddonCard extends React.Component {
  static propTypes = {
    description: PropTypes.string.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      allDescriptionVisible: false,
    };
  }

  toggleDescription = () => {
    this.setState((prevState) => ({
      allDescriptionVisible: !prevState.allDescriptionVisible,
    }));
  };

  get visibleDescription() {
    const { description } = this.props;
    const { allDescriptionVisible } = this.state;

    if (
      allDescriptionVisible
      || description.length <= MAX_DESCRIPTION_VISIBLE_SYMBOLS
    ) {
      return description;
    }

    return `${description.substring(0, MAX_DESCRIPTION_VISIBLE_SYMBOLS)}...`;
  }

  render() {
    return (
      <AddonCardComponent
        {...this.state}
        {...this.props}
        visibleDescription={this.visibleDescription}
        toggleDescription={this.toggleDescription}
      />
    );
  }
}

export default AddonCard;
