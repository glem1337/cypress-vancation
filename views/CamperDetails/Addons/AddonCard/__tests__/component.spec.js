import { shallow } from 'enzyme';

import AddonCard from '../component';

describe('AddonCard component tests', () => {
  const props = {
    iconUrl: 'url',
    price: '50',
    name: 'name',
    priceUnit: 'per day',
    description:
      'description description description description description description',
    visibleDescription: 'descr...',
    allDescriptionVisible: false,
    toggleDescription: jest.fn(),
  };

  const component = shallow(<AddonCard {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
