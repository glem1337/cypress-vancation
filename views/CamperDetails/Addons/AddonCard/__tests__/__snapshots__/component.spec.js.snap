// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddonCard component tests matches snapshot 1`] = `
<div
  className="van-details__adds-card"
>
  <div
    className="d-flex align-items-center justify-content-between mb-20"
  >
    <img
      alt=""
      src="url"
    />
    <Tag
      color="default"
    >
      $50 per day
    </Tag>
  </div>
  <p
    className="text-subtitle font-700 mb-8"
  >
    name
  </p>
  <p>
    descr...
  </p>
  <Button
    onClick={[MockFunction]}
    type="simple-text"
  >
    <Memo(FormattedMessage)
      id="shared.readMore"
    />
  </Button>
</div>
`;
